### DISCUSSION TITLE
Programmatically change display month outside context

### DISCUSSION THREAD

**Discussion by tcolinpa** [gpbl/react-day-picker#1846] [Support] [answered]
How can I control the displayed month outside the date picker context?

Given the demo below, I want to change the displayed month when the user inputs a valid date and selects it (behavior is already on blur or enter keydown). The calendar should display the selected date's month... feels like it lacks a ref.

https://github.com/gpbl/react-day-picker/assets/44402590/49a8e3bc-bc3f-48a2-9e23-3e58e7c54901

Is this possible?

**Comment by gpbl** [comment, marked as answer]
@tcolinpa if I understand correctly, you want the month to be controlled by your component. In this case, you should pass the `month` and `onMonthChange` props, to keep the month in your component's state.

https://react-day-picker.js.org/basics/navigation#controlling-the-current-month

https://codesandbox.io/s/h6pwj5?file=%2FApp.tsx
